.tekton/tasks/deployment-status-report-task.yaml

```diff
@@ -0,0 +1,189 @@
+# SPDX-FileCopyrightText: 2024 SAP edge team
+# SPDX-FileContributor: Manjun Jiao (@mjiao)
+#
+# SPDX-License-Identifier: Apache-2.0
+
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: deployment-status-report
+spec:
+  description: Generate final deployment status report
+  params:
+    - name: aroClusterName
+      type: string
+      description: "ARO cluster name"
+    - name: azureSecretName
+      type: string
+      description: "Name of the Kubernetes Secret containing Azure service principal credentials"
+    - name: deploymentType
+      type: string
+      description: "Type of deployment (e.g., aro-quay-complete)"
+      default: "aro-quay-complete"
+  workspaces:
+    - name: source
+  steps:
+    - name: generate-report
+      image: registry.access.redhat.com/ubi9/ubi
+      workingDir: $(workspaces.source.path)
+      env:
+        - name: CLIENT_ID
+          valueFrom:
+            secretKeyRef:
+              name: $(params.azureSecretName)
+              key: CLIENT_ID
+        - name: CLIENT_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: $(params.azureSecretName)
+              key: CLIENT_SECRET
+        - name: TENANT_ID
+          valueFrom:
+            secretKeyRef:
+              name: $(params.azureSecretName)
+              key: TENANT_ID
+        - name: ARO_RESOURCE_GROUP
+          valueFrom:
+            secretKeyRef:
+              name: $(params.azureSecretName)
+              key: ARO_RESOURCE_GROUP
+        - name: ARO_CLUSTER_NAME
+          value: $(params.aroClusterName)
+        - name: DEPLOYMENT_TYPE
+          value: $(params.deploymentType)
+      script: |
+        #!/bin/bash
+        # Status report should not fail on individual command failures
+        set +e
```

Comment:
this file uses `set +e` while most of the other files use `set -euo pipefail` for ignoring/handling error.